### PR TITLE
[Java] fix default value for JsonNullable

### PR DESCRIPTION
**Pull request by wing328**
to fix https://github.com/OpenAPITools/openapi-generator/issues/14583

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache [hunk 1]
```diff
@@ -96,7 +96,7 @@ public class {{classname}}{{#parent}} extends {{{parent}}}{{/parent}}{{^parent}}
     {{#openApiNullable}}
     {{^required}}
     if (this.{{name}} == null{{#isNullable}} || !this.{{name}}.isPresent(){{/isNullable}}) {
-      this.{{name}} = {{#isNullable}}JsonNullable.of({{{defaultValue}}}){{/isNullable}}{{^isNullable}}{{{defaultValue}}}{{/isNullable}};
+      this.{{name}} = {{#isNullable}}JsonNullable.of({{/isNullable}}{{{defaultValue}}}{{^defaultValue}}new {{#uniqueItems}}LinkedHashSet{{/uniqueItems}}{{^uniqueItems}}ArrayList{{/uniqueItems}}<>(){{/defaultValue}}{{#isNullable}}){{/isNullable}};
```
Comment (welshm) [inline modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:99]: Worth noting the change for `uniqueItems` here - could/should there be a unit test?

**Review by welshm — APPROVED** [review]
Should there be unit tests for this to prevent it from breaking again?

File: modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache [hunk 2]
```diff
@@ -153,7 +148,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
   public {{classname}} add{{nameInCamelCase}}Item({{{items.datatypeWithEnum}}} {{name}}Item) {
     {{#vendorExtensions.x-is-jackson-optional-nullable}}
     if (this.{{name}} == null || !this.{{name}}.isPresent()) {
-      this.{{name}} = JsonNullable.<{{{datatypeWithEnum}}}>of({{{defaultValue}}});
+      this.{{name}} = JsonNullable.<{{{datatypeWithEnum}}}>of({{{defaultValue}}}{{^defaultValue}}new ArrayList<>(){{/defaultValue}});
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache:151]: Why only some libraries have handling for `uniqueItems`?

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache:151]: good point. i think some libraries do not yet support `set` but I could be wrong. 

I'll later push another commit to support set regardless.

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:99]: there's a test: https://github.com/OpenAPITools/openapi-generator/pull/14703/commits/7eb5d5e373ccea95c108fc432a3b67d60c5b25bc#diff-367c850bfdb5159f550162fc5c095292dbe3f083fd7342689640e429d250003eR65

File: modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache]: done via https://github.com/OpenAPITools/openapi-generator/pull/14703/commits/252c7f8cbc3659345c5eb41e6f980cd2e998ff31

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache (hunk 1 above)
Comment (welshm) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:99]: That looks like an updated sample but not a test from the diff?

**Review by welshm — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:99]: sure, tests are always better. will do it in a separate PR with more tests.

**Comment by wing328** [comment]
I've merged it so that users in https://github.com/OpenAPITools/openapi-generator/issues/14583 can test it more easily.

Will file another PR to add more tests.

Thanks for the quick review.

**Pull request merged**